includes/AdminFormSave.php

```diff
@@ -104,8 +104,14 @@ public function save_button_settings( string $form_slug ) { // phpcs:ignore
     public function save_rule_settings( string $form_slug ) { // phpcs:ignore
         $this->update_checkbox( Constants::OUT_OF_STOCK );
         $this->update_checkbox( Constants::MINIMUM_STOCK_THRESHOLD );
-
+        $this->update_checkbox( Constants::ENABLE_TAXONOMY );
         $this->update_number( Constants::BELOW_STOCK_AMOUNT );
+
+        // Inside the save settings method
+        update_option(Constants::ENABLE_TAXONOMY, isset($_POST[Constants::ENABLE_TAXONOMY]) ? Constants::ON : Constants::OFF);
```

Comment:
You not need this line as in line 107 you are already saving checkbox throught `update_checkbo` class method